ingest/src/main/java/com/microsoft/azure/kusto/ingest/IngestionProperties.java

```diff
@@ -230,16 +230,23 @@ public void setIngestIfNotExists(List<String> ingestIfNotExists) {
         return fullAdditionalProperties;
     }
 
-    public void setDataFormat(DataFormat dataFormat) {
-        this.dataFormat = dataFormat;
+    /**
+     * Sets the data format. If null is passed, then it does not set it.
+     *
+     * @param dataFormat One of the values in: {@link DataFormat DataFormat}
+     */
+    public void setDataFormat(@NotNull DataFormat dataFormat) {
+        if (dataFormat != null) {
```

Comment:
@NotNull means that we expect to never get null as a parameter (usually throws if we got it). If we handle it then the correct annotation is @CanBeNull

Reply:
We don't allow them to pass null here.

Reply:
More accurately, they can technically send null, but it's pointless.

Reply:
I disagree, we allow them to pass null, since the method explicitly supports it. 
I think you should throw if it's null, having it do nothing is a surprising behaviour.